UnityProject/Assets/Scripts/UI/Tooltip.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+public class Tooltip : MonoBehaviour, IPointerEnterHandler, IPointerExitHandler
+{
+    private static GameObject tooltipResource = null;
+    const float TOOLTIP_INTERVAL = 1.0f;
+    private float enterTime = 0;
+    private GameObject tooltipObject = null;
+    [SerializeField]
+    private string tooltipText = "";
+
+    void Start()
+    {
+        if (tooltipResource == null) {
+            tooltipResource = Resources.Load<GameObject>("UI/GUI/Tooltip");
+        }
+    }
+
+    void Update()
+    {
+        if (tooltipObject) {
+            // Move tooltip to mouse
+            tooltipObject.transform.position = Input.mousePosition - new Vector3(0, 20, 0);
+        } else if (enterTime != 0 && Time.realtimeSinceStartup - enterTime > TOOLTIP_INTERVAL) { 
+            // Create tooltip if needed
+            tooltipObject = Instantiate(tooltipResource, Vector3.zero, Quaternion.identity);
```

Comment:
you dont have to constantly create and destroy this object, just reuse it

Reply:
yep when you are done with it just set the toolTipObject.SetActive(false) and turn it back on when you need it